twine/commands/check.py

```diff
@@ -143,6 +148,7 @@ def check(dists: List[str], output_stream: IO[str] = sys.stdout) -> bool:
 
 def main(args: List[str]) -> bool:
     parser = argparse.ArgumentParser(prog="twine check")
+    settings.Settings.register_argparse_arguments(parser)
```

Comment:
@bhrutledge How do you propose we share commands like `--no-color` between multiple sub-commands instead? In https://github.com/pypa/twine/issues/659, @sigmavirus24 said that the `Settings` object should be used for this.

Reply:
So far we've needed thse arguments for all the commands and `check` was implemented while I was not paying attention. Otherwise, we'd have resolved this sooner. I think separate classmethods for things that don't want need all of those arguments is the way to go. Then `register_argparse_arguments` can call those and then add its own arguments.

Reply:
I agree that `check` sort of skirted this issue and should eventually be fixed.

> I think separate classmethods for things that don't want need all of those arguments is the way to go.

Can you elaborate on this? This kind of sounds like what https://github.com/pypa/twine/issues/659 could solve with class composition/inheritance. Do you mean extra classmethods on the `Settings` class instead? Something like `Settings._add_register_and_upload_arguments`?

Reply:
> Can you elaborate on this?

Sure.

> This kind of sounds like what #659 could solve with class composition/inheritance.

Except that the TwineCommand class would never be part of the public API the way Settings would be as an escape hatch from the high-level porcelain API planned.

I'm thinking of breaking up registration into class methods based on relationship:

```py
Settings.register_repository_arguments(...)
Settings.register_gpg_arguments(...)
Settings.register_identity_arguments(...)
```

And those are called by

```py
class Settings:
    @classmethod
    def register_upload_arguments(cls, ...):
        cls.register_repository_arguments(...)
        cls.register_gpg_arguments(...)
        cls.register_identity_arguments(...)
        # plus anything specific to upload
```

Likewise for check and register. I doubt register should have `--comment` or `--skip-existing`, so those would be specific to upload. This would give us logical groupings that are more explicit than what we have no and allow commands to register what they thematically need.

Reply:
@sigmavirus24 would you like @callmecampos to move forward with that in this PR? Should it come before this PR?

Reply:
I would be worried about that blocking this PR. As a transitory state, we could have `check` use all the options and in @callmecampos's next PR they could "fix" it.

Reply:
Real quick: I have an intermediate idea that I'd like to experiment with. Can @callmecampos wait a day?

Reply:
I've just opened https://github.com/callmecampos/twine/pull/2 with my idea, which a screenshot and passing tests, but it's basically:

```
$ twine --no-color upload dist/*
```

Since that reverts some changes in this PR, it might be easier to review the overall diff: https://github.com/pypa/twine/compare/master...bhrutledge:red-text-error

Reply:
Re: adding class methods to `Settings`, I support that in theory, but not as part of this PR. I also note that there are number of [pending issues](https://github.com/pypa/twine/projects/2) to improve `check`, which makes me wonder if the `Settings` restructuring could be part of that work, and maybe released as Twine 4.0, to give us some flexibility on the API.

Reply:
> and maybe released as Twine 4.0, to give us some flexibility on the API.

Why as 4.0? Also perhaps this isn't the best place to have this long-winded discussion

Reply:
I'm also worried about how we're approaching this. We don't seem focused or aligned on a goal right now and instead we're bandying about solutions and focusing on those.

What are our goals and constraints for the solution beyond "Can disable color?"